components/skip-link/src/index.tsx

```diff
@@ -29,9 +30,43 @@ export const SkipLink = styled('a')(
   }
 );
 
-SkipLink.defaultProps = {
-  children: 'Skip to main content',
-  href: '#content',
+interface SkipLinkOwnProps extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
+  children?: string | React.ReactNode;
+  href?: string;
+}
+
+type SkipLinkRefType = React.Ref<HTMLAnchorElement>;
+
+export const SkipLink: SkipLinkType = React.forwardRef(
+  ({ children = 'Skip to main content', href = '#content', ...props }: SkipLinkOwnProps, ref: SkipLinkRefType) => (
+    <StyledSkipLink ref={ref} href={href} {...props}>
+      {children}
+    </StyledSkipLink>
+  )
+);
+
+export interface SkipLinkType extends React.ForwardRefExoticComponent<SkipLinkOwnProps> {
+  (props: SkipLinkPropsWithoutAs, ref?: SkipLinkRefType): React.ReactElement<SkipLinkPropsWithoutAs>;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  <AsC extends string | React.ComponentType<any> = 'a', FAsC extends string | React.ComponentType<any> = AsC>(
+    props: SkipLinkPropsWithAs<AsC, FAsC>,
+    ref?: React.Ref<AsC>
+  ): React.ReactElement<SkipLinkPropsWithAs<AsC, FAsC>>;
+}
+
+type SkipLinkPropsWithoutAs = StyledComponentProps<'a', never, SkipLinkOwnProps, never> & {
+  as?: never | undefined;
+  forwardedAs?: never | undefined;
+};
+
+type SkipLinkPropsWithAs<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  AsC extends string | React.ComponentType<any>,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  FAsC extends string | React.ComponentType<any> = AsC
+> = StyledComponentProps<AsC, never, SkipLinkOwnProps, never, FAsC> & {
```

Comment:
I think we can remove the `any` by doing something like

```suggestion
type SkipLinkPropsWithAs<
  Props,
  AsC extends string | React.ComponentType< Props >,
  FAsC extends string | React.ComponentType< Props > = AsC
> = StyledComponentProps<AsC, never, SkipLinkOwnProps, never, FAsC> & {
```

but I guess this was copied from elsewhere, this isn't high priority, and perhaps something that should be addressed across all places it's done in its own PR

Reply:
Yes, it was copied. I think it's best addressed in it's own PR.